src/components/RoomCard.tsx

```diff
@@ -118,11 +134,13 @@ export const RoomCard: React.FC<{
           </Text>
           <Button
             label={'Book for $$$ xDAI'}
-            onClick={() => {
-              handleBook();
-            }}
+            onClick={() => handleBook()}
+            icon={loading ? <Spinner /> : undefined}
```

Comment:
You can also do `loading && <Spinner/>`

A bit less code

Reply:
![Screenshot 2022-08-04 at 15 23 56](https://user-images.githubusercontent.com/47505482/182857913-db1e909b-9fa4-49e2-9f85-9cf4f7913bfe.jpg)